### DISCUSSION TITLE
Adding a Watch expression using AutoKey (documentation suggestion)

### DISCUSSION THREAD

**Discussion by mnieber** [inducer/pudb#583] [Ideas]
I just figured out a fast way to add variables to the Watch section, using AutoKey. I don't know if this is of interest, but the following script adds the current selection as a new Watch entry (perhaps it's worth adding this to the documentation as a productivity hint)

```
import time
def pause():
    time.sleep(0.1)
    
text = clipboard.get_selection()
keyboard.send_key('<right>')
pause()
keyboard.send_key('<insert>')
pause()
keyboard.send_keys(text)
pause()
keyboard.send_key('<enter>')
pause()
keyboard.send_key('<left>')
```

**Comment by inducer** [comment]
Thanks for pointing this out. I'm not totally sure what to do with this, since it goes quite a ways beyond pudb's current UI paradigm, and it is, for better or worse, a bit brittle and bolted-on (not a criticism), so I don't know if I'd like to encourage it. :slightly_smiling_face: 

That said, there are some valid questions that arise out of this:
- What's a good UI for selecting variable names/other things in the source code? (Vimium-style? Urwid does have mouse support, even if we don't do very much with it ATM?)
- How can we better support user-driven automation? ("plugins"?) Should we?

I think I'll push this over to discussions for the time being.

**Reply by mnieber** [reply]
I agree, using Autokey is a hack :-).

Part of a good UI could be to emulate mouse hovering, e.g.:

- switch to hover mode
- when you click somewhere in the code, then the value of clicked variable is shown in the side bar (but no watch is added, though maybe [enter] would create a watch from the "hovered" variable)
- if you want to "hover" an expression, then you make a text-selection with the mouse
